### PR TITLE
feat: Adding calls to custom style hooks derived from context in react-combobox components

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Updated components to call custom style hook

## Issues
- Updates #27058

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                      Change |
| :---------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `85.509 kB`<br />`27.437 kB` | `88.434 kB`<br />`28.035 kB` | `2.925 kB` <br />`598 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>                    | `81.926 kB`<br />`27.213 kB` | `84.756 kB`<br />`27.754 kB` |  `2.83 kB` <br />`541 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `84.196 kB`<br />`27.191 kB` | `87.121 kB`<br />`27.773 kB` | `2.925 kB` <br />`582 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.061 kB`<br />`18.105 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `205.644 kB`<br />`57.056 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `37.891 kB`<br />`12.039 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.324 kB`<br />`2.129 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/efb2274e37c0237f3a10a4630b2696a9880f6398'>efb2274e37c0237f3a10a4630b2696a9880f6398</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/Avatar-mount_base.html">1292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/Avatar-mount.html">1311</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/Button-mount_base.html">969</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/Button-mount.html">959</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/Field-mount_base.html">2079</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/Field-mount.html">2065</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/FluentProvider-mount_base.html">1657</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/FluentProvider-mount.html">1721</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/FluentProviderWithTheme-mount_base.html">646</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/FluentProviderWithTheme-mount.html">655</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">591</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">648</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">645</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/InfoButton-mount_base.html">604</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/InfoButton-mount.html">559</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/MakeStyles-mount_base.html">1950</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/MakeStyles-mount.html">1934</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/Persona-mount_base.html">3065</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/Persona-mount.html">3074</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/SpinButton-mount_base.html">2513</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27063/perf-test-react-components/SpinButton-mount.html">2512</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: change/@fluentui-react-combobox-e0e79789-18c1-4eea-bc98-3731d6751866.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat: custom styles",
```
Comment (khmakoto) [inline change/@fluentui-react-combobox-e0e79789-18c1-4eea-bc98-3731d6751866.json:3]:
```suggestion
  "comment": "feat: Adding calls to custom style hooks derived from context.",
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**